src/esp/io/JsonEspTypes.h

```diff
@@ -0,0 +1,161 @@
+  frame = esp::geo::CoordinateFrame(up, front, origin);
+  return true;
+}
+
+inline JsonGenericValue toJsonValue(const esp::assets::AssetInfo& x,
+                                    JsonAllocator& allocator) {
+  JsonGenericValue obj(rapidjson::kObjectType);
+  addMemberAsUint32(obj, "type", x.type, allocator);
+  addMember(obj, "filepath", x.filepath, allocator);
+  addMember(obj, "frame", x.frame, allocator);
+  addMember(obj, "virtualUnitToMeters", x.virtualUnitToMeters, allocator);
+  addMember(obj, "requiresLighting", x.requiresLighting, allocator);
+  addMember(obj, "splitInstanceMesh", x.splitInstanceMesh, allocator);
+  return obj;
+}
+
+inline bool fromJsonValue(const JsonGenericValue& obj,
+                          esp::assets::AssetInfo& x) {
+  readMemberAsUint32(obj, "type", x.type);
+  readMember(obj, "filepath", x.filepath);
+  readMember(obj, "frame", x.frame);
+  readMember(obj, "virtualUnitToMeters", x.virtualUnitToMeters);
+  readMember(obj, "requiresLighting", x.requiresLighting);
+  readMember(obj, "splitInstanceMesh", x.splitInstanceMesh);
+  return true;
+}
+
+inline JsonGenericValue toJsonValue(
+    const esp::assets::RenderAssetInstanceCreationInfo& x,
+    JsonAllocator& allocator) {
+  JsonGenericValue obj(rapidjson::kObjectType);
+  addMember(obj, "filepath", x.filepath, allocator);
+  addMember(obj, "scale", x.scale, allocator);
+  addMember(obj, "isStatic", x.isStatic(), allocator);
+  addMember(obj, "isRGBD", x.isRGBD(), allocator);
+  addMember(obj, "isSemantic", x.isSemantic(), allocator);
+  addMember(obj, "lightSetupKey", x.lightSetupKey, allocator);
+  return obj;
+}
+
+inline bool fromJsonValue(const JsonGenericValue& obj,
+                          esp::assets::RenderAssetInstanceCreationInfo& x) {
+  readMember(obj, "filepath", x.filepath);
+  readMember(obj, "scale", x.scale);
+  bool isStatic = false;
+  readMember(obj, "isStatic", isStatic);
+  bool isRGBD = false;
+  readMember(obj, "isRGBD", isRGBD);
+  bool isSemantic = false;
+  readMember(obj, "isSemantic", isSemantic);
+  if (isStatic) {
+    x.flags |= esp::assets::RenderAssetInstanceCreationInfo::Flag::IsStatic;
+  }
+  if (isRGBD) {
+    x.flags |= esp::assets::RenderAssetInstanceCreationInfo::Flag::IsRGBD;
+  }
+  if (isSemantic) {
+    x.flags |= esp::assets::RenderAssetInstanceCreationInfo::Flag::IsSemantic;
+  }
+  readMember(obj, "lightSetupKey", x.lightSetupKey);
+  return true;
+}
+
+inline JsonGenericValue toJsonValue(const esp::gfx::replay::Transform& x,
+                                    JsonAllocator& allocator) {
+  JsonGenericValue obj(rapidjson::kObjectType);
+  addMember(obj, "translation", x.translation, allocator);
+  addMember(obj, "rotation", x.rotation, allocator);
+  return obj;
+}
+
+inline bool fromJsonValue(const JsonGenericValue& obj,
+                          esp::gfx::replay::Transform& x) {
+  readMember(obj, "translation", x.translation);
+  readMember(obj, "rotation", x.rotation);
+  return true;
+}
```

Comment:
Here as well, I would like to have an accurate return value. (I could see a useful application of these transformations in the scene instance attributes, but I would want to have an accurate state of the read value).  So something like this :

```
bool success = true;
success &= readMember(obj, "translation", x.translation);
success &= readMember(obj, "rotation", x.rotation);
return success;
```

Reply:
Sure.

>I could see a useful application of these transformations in the scene instance attributes,
Beware, this class in in the replay namespace for a reason and might change at the whims of what replay needs, e.g. adding scale, or switching from quaternion to Matrix4. I don't recommend for general usage outside replay.

Reply:
I see, that makes sense.  Thanks for the clarification.

Reply:
Might be nice to have a generic transformation read/write path, but that doesn't have to be in this PR. :).